src/Orleans.Core/Transactions/TransactionContext.cs

```diff
@@ -13,18 +11,18 @@ public static class TransactionContext
         internal const string TransactionInfoHeader = "#TC_TI";
         internal const string Orleans_TransactionContext_Key = "#ORL_TC";
 
-        public static TransactionInfo GetTransactionInfo()
+        public static ITransactionInfo GetTransactionInfo()
```

Comment:
Upon further reflection, since we're allowing ITransactionInfo to be a general interface, but specific systems will need aspecific implementation (which seems to be the current pattern) maybe the context can have a GetTransactionInfo<T>() where T : ITransactionInfo call which returns null if the specific type of ITransactionInfo has not been set.  Then all of the related systems need only perform null checks (which they 'should' be doing anyway, but are not).

Reply:
I have added an overload GetRequiredTransactionInfo<T>() which throws an informative exception if the info is null or the wrong type.